libs/image-utils/src/dither.ts

```diff
@@ -0,0 +1,87 @@
+import { ok, Result } from '@votingworks/types';
+import { assert } from '@votingworks/utils';
+import { createImageData } from 'canvas';
+import {
+  getImageChannelCount,
+  ImageProcessingError,
+  isGrayscale,
+  isRgba,
+  loadImageData,
+  toGrayscale,
+  toRgba,
+  writeImageData,
+} from './image_data';
+
+/**
+ * Dithers an ImageData using the Floyd-Steinberg algorithm. Expects a grayscale
+ * image and returns a monochrome image.
+ */
+export function dither(
+  input: ImageData
+): Result<ImageData, ImageProcessingError> {
+  if (!isGrayscale(input)) {
+    const toGrayscaleResult = toGrayscale(input);
+    if (toGrayscaleResult.isErr()) {
+      return toGrayscaleResult;
+    }
+    return dither(toGrayscaleResult.ok());
+  }
+
+  const channelCount = 1;
+  const { data, width, height } = input;
+  const src = Float32Array.from(data);
+  const dst = new Uint8ClampedArray(width * height * channelCount);
+  const output = createImageData(dst, width, height);
+  const threshold = 128;
```

Comment:
Have you played with other thresholds? I have found that, when just doing template mapping, 128 is not the right threshold and it should be higher.

Reply:
I added `threshold` as an option in 21b890b and made the default use `otsu`. The results don't change that much, though:

| 50 | 100 | 200 | otsu (110) |
|-|-|-|-|
|![mona-lisa-dithered50](https://user-images.githubusercontent.com/1938/206264799-8bbc6ef9-b22d-4fa3-8623-ce2103a4394c.png)|![mona-lisa-dithered100](https://user-images.githubusercontent.com/1938/206264833-83659013-8dc8-4c9d-bec4-1e9912dc0a63.png)|![mona-lisa-dithered200](https://user-images.githubusercontent.com/1938/206264862-33d9c6b2-67b7-47f9-b6b7-6b39ea730660.png)|![mona-lisa-dithered](https://user-images.githubusercontent.com/1938/206264878-b5202bd8-a4fb-4dae-80ec-cbc29b1832a2.png)|